### PR TITLE
Split RGB_MATRIX_ENABLE into _ENABLE and _DRIVER

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

~~It's probably about time we deprecated this.~~

Heads up to keymap owners:
~~@drashna - `crkbd`, `layouts/ortho_4x12`~~
~~@bcat - `crkbd`~~
@Curry - `crkbd`
@dsanchezseco - `crkbd`
@kidBrazil - `crkbd`
@ninjonas - `crkbd`
~~@rpbaptist - `crkbd`~~
~~@soundmonster - `crkbd`~~
@Salicylic-acid3 - `naked48`, `setta21`
~~@kageurufu - `rgbkb/sol`~~
~~@XScorpion2 - `rgbkb/sol`~~
~~@toshi0383 - `zinc`~~
~~@tominabox1 - userspace~~

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]
LGTM

**Review by noroadsleft — APPROVED** [review]
Looks good, pending user approval.

**Review by zvecr — COMMENTED** [review]
Personally i would rather have and `enable = yes/no` and `driver = IS31blah`.

**Review by zvecr — APPROVED** [review]

**Comment by zvecr** [comment]
image

**Review by drashna — APPROVED** [review]

**Review by XScorpion2 — APPROVED** [review]
meh, don't understand why 2 defines is better than 1, but approved anyways.

**Comment by fauxpark** [comment]
@XScorpion2 
The idea is that users should not need to know which specific driver is used on a given board to turn RGB matrix on if it is supported but not enabled by default.

**Comment by zvecr** [comment]
I guess there was a little context missing in my previous comment. Having the 2 parts of the logic split has a few benefits,
1. Configurator can just end up with a enable/disable toggle switch without having to know what driver is the correct one to specify
2. Where not enabled by default (rev6 planck is a good example), it allows users to not have to dig for the correct driver setting
3. Userspace "disable rgb matrix by default, but enable it on this single board" only has to interact with the enable flag

**Comment by XScorpion2** [comment]
Thanks for the extra context, coffee hit too which helps

**Review by noroadsleft — COMMENTED** [review]
Based on `users/tominabox1/rules.mk`, it seems `keyboards/lazydesigners/dimple/rules.mk` should have:

```makefile
RGB_MATRIX_ENABLE = no
RGB_MATRIX_DRIVE = WS2812
```

**Review by tominabox1 — COMMENTED** [review]
the Dimple does not support rgb_matrix in the default sense. I was planning on working on it some time ago and never did it. Shouldn't add this to the keyboard folder. 

Would suggest removing the RGB_Matrix stuff from the Dimple section of my userspace rules.mk.  

I approve the change for CRKBD in my rules.

**Review by Erovia — APPROVED** [review]
LGTM

**Review by kageurufu — APPROVED** [review]

**Comment by tominabox1** [comment]
looks good

**Review by noroadsleft — APPROVED** [review]

**Comment by bcat** [comment]
Not sure if this is blocked on LGTMs from each user, but LGTM for my keymap. (Is there some way to declare "it's fine to refactor my keymaps/userspace; I'll fix them later as needed"?)

**Comment by rpbaptist** [comment]
> Not sure if this is blocked on LGTMs from each user, but LGTM for my keymap. (Is there some way to declare "it's fine to refactor my keymaps/userspace; I'll fix them later as needed"?)

Same for me. Just go ahead. I'll just update my keymap afterwards.

**Comment by elfmimi** [comment]
I'd like to point out that this is great for keyboards which have indicator LEDs as part of the led-matrix .
RGB effects can be disabled in the expected way and still have indicator LEDs functional.

**Comment by drashna** [comment]
looks like this needs a rebase

**Review by soundmonster — APPROVED** [review]
👍

File: keyboards/crkbd/keymaps/soundmonster/rules.mk
Comment (soundmonster) [inline keyboards/crkbd/keymaps/soundmonster/rules.mk:2]: I guess the driver should be specified as well?
```suggestion
RGB_MATRIX_ENABLE  = yes
RGB_MATRIX_DRIVER   = WS2812
```

Comment (fauxpark) [reply keyboards/crkbd/keymaps/soundmonster/rules.mk:2]: Nope, it's specified in the keyboard-level rules.mk now (that's the whole point of this PR).

**Comment by drashna** [comment]
Has some merge conflicts

**Comment by fauxpark** [comment]
I think we may need one more rebase of develop so I can include more recent boards which have RGB Matrix, so that we don't break them.

**Review by noroadsleft — APPROVED** [review]

**Pull request merged**
